app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -117,6 +124,13 @@ private class HomeTileAdapter(val onUrlEnteredListener: OnUrlEnteredListener, ho
             onUrlEnteredListener.onNonTextInputUrlEntered(item.url)
             TelemetryWrapper.homeTileClickEvent()
         }
+        itemView.setOnLongClickListener {
+            deleteTile(item.id)
+            tiles.removeAt(position)
+            notifyItemRemoved(position)
+            notifyItemRangeChanged(position, tiles.size)
```

Comment:
Do you need this notifyItemRangeChanged?

Reply:
Yes or else it crashes when I remove something from the [0,n-1) range and then n-1. Unless there's another way to notify the elements that their index has changed??

Reply:
Mmmmm I'm pretty sure you don't need to call that, because in the documentation you should just be able to call "notifyItemRemoved". I'm currently trying to finish up pinning so I don't have a ton of time to try to debug this with you, but maybe try the following:

- Make sure the data source is updating correctly (is getItemCount returning the correct length?)
- Look at other existing code in Fennec ( https://dxr.mozilla.org/mozilla-central/source/mobile/android/base/java/org/mozilla/gecko/tabs/TabStripAdapter.java#74 or https://dxr.mozilla.org/mozilla-central/search?q=notifyitemremoved&redirect=true )
- Take a look at the docs ( https://developer.android.com/reference/android/support/v7/widget/RecyclerView.Adapter.html )
- See if other people have this problem on stackoverflow?

If you still can't figure this out, we can take a look at this together.